src/Pages/Dashboard/Sensor.jsx

```diff
@@ -1,62 +1,118 @@
-import React, { PureComponent } from "react";
+import React, {PureComponent} from "react";
 // Material UI Components
-import { withStyles } from "@material-ui/core";
+import {withStyles} from "@material-ui/core";
 //Style
-import { dashboardStyle } from "./dashboardStyle";
+import {dashboardStyle} from "./dashboardStyle";
 import Dashboard from "./Dashboard";
 
 class Sensor extends PureComponent {
-  constructor(props) {
-    super(props);
-    this.db = null;
-    this.state = {
-      data: []
+    constructor(props) {
+        super(props);
+        this.db = null;
+        this.state = {
+            data: [],
+            historydata: []
+        };
+        let date = new Date();
+        let now = date.getTime();
+        let d = now - (30 * 24 * 60 * 60 * 1000);
+        this.timestamp = Math.floor(d / 1000);
+        this.nowstamp = Math.floor(now / 1000);
+    }
+
+    // We wait until the component has been mounted (which means
+    // other components are unmounted) before polling, so we don't
+    // have multiple registers to a particular database
+    componentDidMount = () => {
+        this.getDatabase(this.props.sensorId);
+    };
+
+    changeRange = (value) => {
+        this.setState({selected: value});
+        let offset;
+        if (value === 1) {
```

Comment:
I would probably extract these out into a map, just so they aren't hard-coded magic values in the code itself.

For example, you could do something like this at the top of the file as a constant:
```javascript
const offsetMap = {
    1: 60 * 60 * 1000,
    2: 2 * 60 * 60 * 1000,
    ...
};
```

Then use it like:
```javascript
offset = timeMap[value];
```